src/m365/teams/commands/meeting/meeting-add.ts

```diff
@@ -0,0 +1,216 @@
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        startTime: typeof args.options.startTime !== 'undefined',
+        endTime: typeof args.options.endTime !== 'undefined',
+        subject: typeof args.options.subject !== 'undefined',
+        participantUserNames: typeof args.options.participantUserNames !== 'undefined',
+        organizerEmail: typeof args.options.organizerEmail !== 'undefined',
+        recordAutomatically: !!args.options.recordAutomatically
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-s, --startTime [startTime]'
+      },
+      {
+        option: '-e, --endTime [endTime]'
+      },
+      {
+        option: '--subject [subject]'
+      },
+      {
+        option: '-p, --participantUserNames [participantUserNames]'
+      },
+      {
+        option: '--organizerEmail [organizerEmail]'
+      },
+      {
+        option: '-r, --recordAutomatically'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+
+        if (args.options.startTime && !validation.isValidISODateTime(args.options.startTime)) {
+          return `'${args.options.startTime}' is not a valid ISO date string for startTime.`;
+        }
+
+        if (args.options.endTime && !validation.isValidISODateTime(args.options.endTime)) {
+          return `'${args.options.endTime}' is not a valid ISO date string for endTime.`;
+        }
+
+        if (args.options.startTime && args.options.endTime && new Date(args.options.startTime) >= new Date(args.options.endTime)) {
+          return 'The startTime value must be before endTime.';
+        }
+
+        if (args.options.endTime && !args.options.startTime && new Date() >= new Date(args.options.endTime)) {
+          return 'When only the endTime is specified, it needs to be after the current time.';
+        }
+
+        if (args.options.participantUserNames) {
+
+          if (args.options.participantUserNames.indexOf(',') === -1 && !validation.isValidUserPrincipalName(args.options.participantUserNames)) {
+            return `${args.options.participantUserNames} contains invalid UPN.`;
+          }
+
+          const participants = args.options.participantUserNames.trim().toLowerCase().split(',').filter(e => e && e !== '');
+
+          if (!participants || participants.length === 0 || participants.some(e => !validation.isValidUserPrincipalName(e))) {
+            return `${args.options.participantUserNames} contains one or more invalid UPN.`;
+          }
+        }
+
+        if (args.options.organizerEmail && !validation.isValidUserPrincipalName(args.options.organizerEmail)) {
+          return `'${args.options.organizerEmail}' is not a valid email for organizerEmail.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  /**
```

Comment:
This comment on the commandAction method may be removed.

Reply:
Done